test/jdk/javax/swing/JFrame/bug4419914.java

```diff
@@ -28,55 +28,143 @@
  * @library /java/awt/regtesthelpers
  * @build PassFailJFrame
  * @run main/manual bug4419914
-*/
+ */
 
 import java.awt.BorderLayout;
 import java.awt.ComponentOrientation;
+import java.awt.Frame;
+import java.awt.Window;
+import java.util.List;
+import java.util.Locale;
 import javax.swing.JButton;
+import javax.swing.JDialog;
 import javax.swing.JFrame;
-import java.util.Locale;
+import javax.swing.JLabel;
+import javax.swing.JWindow;
 
 public class bug4419914 {
+    private static JFrame frame;
     private static final String INSTRUCTIONS = """
-        1. You will see a frame with five buttons.
-        2. Confirm that each button is placed as follows:
-             NORTH
-        END  CENTER  START
-             SOUTH
-        3. Press the "NORTH" button and confirm the button is focused.
-        4. Press TAB repeatedly and confirm that the TAB focus moves from right to left.
-             (NORTH - START - CENTER - END - SOUTH - NORTH - START - CENTER - ...)
+        This test verifies tab movement on RTL component orientation
+        in JWindow, JFrame and JDialog.
+
+        When test starts 3 test windows are displayed - JFrame, JWindow and JDialog.
+        Follow the instructions below and if any condition does not hold
+        press FAIL.
+
+        1. Confirm that each button in the child window is placed as follows:
+
+            For JFrame:
+                 NORTH
+            END  CENTER  START
+                 SOUTH
+
+            For JWindow:
+            END  CENTER  START
+                  QUIT
+
+            For JDialog:
+            END  CENTER  START
+
+        3. Press on the "START" button in case JWindow & JDialog and "NORTH"
+           in case of JFrame, confirm that the respective button is focused.
+
+        4. Press TAB repeatedly and confirm that the TAB focus moves
+           from right to left.
+
+            For JFrame:
+            (NORTH - START - CENTER - END - SOUTH - NORTH - START - CENTER - ...)
 
-            If there's anything different from the above items, click Fail else click Pass.""";
+            For JWindow:
+            (START - CENTER - END - Quit - START - CENTER - END - Quit - ...)
+
+            For JDialog:
+            (START - CENTER - END - START - CENTER - END - ...)
+
+        If all of the above conditions are true press PASS else FAIL.
+        """;
 
     public static void main(String[] args) throws Exception {
         PassFailJFrame.builder()
                 .title("Tab movement Instructions")
                 .instructions(INSTRUCTIONS)
                 .rows((int) INSTRUCTIONS.lines().count() + 2)
-                .columns(48)
-                .testUI(bug4419914::createTestUI)
+                .columns(45)
+                .testTimeOut(15)
+                .testUI(bug4419914::createAndShowUI)
+                .positionTestUI(WindowLayouts::rightOneRow)
```

Comment:
I find `WindowLayouts::rightOneColumn` suitable for the testUIs alignment. Test windows are aligned vertically which looks better than horizontal alignment.